compat/src/suspense.js

```diff
@@ -0,0 +1,143 @@
+import { Component, createElement, unmount } from 'preact';
+import { removeNode } from '../../src/util';
+
+export function catchRender(error, component) {
+	// thrown Promises are meant to suspend...
+	if (error.then) {
+		for (; component; component = component._ancestorComponent) {
```

Comment:
Can a `while(component = component._ancestorComponent)` do it in here? Maybe we can shave some bytes